app/models/work_completion.rb

```diff
@@ -42,8 +42,28 @@ class WorkCompletion < ApplicationRecord
 
   private
 
+  # @!method pass_and_attach_requests
+  #   Determines the appropriate processing class based on the target labware type
+  #   and invokes its `process` method to handle request passing and linking.
+  #
+  #   @return [void]
+  #
+  #   @example
+  #     pass_and_attach_requests
+  #
+  #   The method selects the processing class as follows:
+  #   - If the target labware responds to `racked_tubes`, it uses `TubeRackCompletion`.
+  #   - If the target labware responds to `wells`, it uses `PlateCompletion`.
+  #   - Otherwise, it defaults to `TubeCompletion`.
   def pass_and_attach_requests
-    processing_class = target.respond_to?(:wells) ? PlateCompletion : TubeCompletion
+    processing_class =
+      if target.respond_to?(:wells)
+        PlateCompletion
+      elsif target.respond_to?(:racked_tubes)
+        TubeRackCompletion
+      else
```

Comment:
I'm not sure I like how the existing code just assumes it must be a tube if it's not a plate with wells. Seems a bit risky to me, would rather it positively checked it's a tube and throw exception if target doesn't match something in the list.
Not sure what there is unique about a tube though... a tube is also a receptacle maybe.